assert_allclose reports NaN as max relative difference if arrays have zeros at the same positions

<!-- Please describe the issue in detail here, and fill in the fields below -->
If two different arrays share a zero at the same position the max relative difference will be reported as NaN by ```assert_allclose```. This is because in ```assert_array_compare``` the error is simply calculated as ```max_rel_error = (error / abs(y)).max()``` which can result in 0/0 = NaN. 
This hides the actual biggest relative difference and makes it more difficult to solve the problem in your code that is triggering the assertion. 
### Reproducing code example:

```python
import numpy as np
x = np.array([0, 1])
y = np.array([0, 2])
np.testing.assert_allclose(x, y, rtol=0.001)
```

### Output:
```
AssertionError: 
Not equal to tolerance rtol=0.001, atol=0

Mismatch: 50%
Max absolute difference: 1
Max relative difference: nan
 x: array([0, 1])
 y: array([0, 2])
```

### Expected output:
```
AssertionError: 
Not equal to tolerance rtol=0.001, atol=0

Mismatch: 50%
Max absolute difference: 1
Max relative difference: 0.5
 x: array([0, 1])
 y: array([0, 2])
```

### Numpy/Python version information:
1.16.4

--- a/numpy/testing/_private/utils.py
+++ b/numpy/testing/_private/utils.py
@@ -703,7 +703,7 @@ def assert_array_compare(comparison, x, y, err_msg='', verbose=True,
                          header='', precision=6, equal_nan=True,
                          equal_inf=True):
     __tracebackhide__ = True  # Hide traceback for py.test
-    from numpy.core import array, array2string, isnan, inf, bool_, errstate
+    from numpy.core import array, array2string, isnan, inf, bool_, errstate, all
 
     x = array(x, copy=False, subok=True)
     y = array(y, copy=False, subok=True)
@@ -821,7 +821,12 @@ def assert_array_compare(comparison, x, y, err_msg='', verbose=True,
 
                     # note: this definition of relative error matches that one
                     # used by assert_allclose (found in np.isclose)
-                    max_rel_error = (error / abs(y)).max()
+                    # Filter values where the divisor would be zero
+                    nonzero = bool_(y != 0)
+                    if all(~nonzero):
+                        max_rel_error = array(inf)
+                    else:
+                        max_rel_error = (error[nonzero] / abs(y[nonzero])).max()
                     if error.dtype == 'object':
                         remarks.append('Max relative difference: '
                                         + str(max_rel_error))

--- a/numpy/testing/tests/test_utils.py
+++ b/numpy/testing/tests/test_utils.py
@@ -880,6 +880,15 @@ class TestAssertAllclose(object):
         assert_array_less(a, b)
         assert_allclose(a, b)
 
+    def test_report_max_relative_error(self):
+        a = np.array([0, 1])
+        b = np.array([0, 2])
+
+        with pytest.raises(AssertionError) as exc_info:
+            assert_allclose(a, b)
+        msg = str(exc_info.value)
+        assert_('Max relative difference: 0.5' in msg)
+
 
 class TestArrayAlmostEqualNulp(object):
 

